edb/pgsql/ast.py

```diff
@@ -667,10 +667,7 @@ class Expr(ImmutableBaseExpr):
 
 
 class BaseConstant(ImmutableBaseExpr):
-    def __init__(self, **kwargs):
-        super().__init__(**kwargs)
-        if not isinstance(self, NullConstant) and self.val is None:
-            raise ValueError('cannot create a pgast.Constant without a value')
+    pass
```

Comment:
Again, `.replace()` calls __init__(), and then copies the fields across, so we lose our pretty safety check.